src/compiler/checker.ts

```diff
@@ -16183,9 +16256,19 @@ namespace ts {
             return getSignatureInstantiation(signature, getInferredTypes(context), isInJavaScriptFile(contextualSignature.declaration));
         }
 
-        function inferJsxTypeArguments(signature: Signature, sourceAttributesType: Type, context: InferenceContext): Type[] {
-            const paramType = getTypeAtPosition(signature, 0);
-            inferTypes(context.inferences, sourceAttributesType, paramType);
+        function inferJsxTypeArguments(signature: Signature, node: JsxOpeningLikeElement, context: InferenceContext): Type[] {
+            {

```

Comment:
If using a block like this isn't a lint violation already, it should be.

Is it actually unsafe to reuse `paramType` ?

Reply:
The blocks are just there for organization (as in `inferTypeArguments` each pass is an identifiable logical block thanks to each cycle of argument iteration), not to allow reuse of the same name safely. I'll remove them, if you'd prefer.